test/03_reduction/reduction_resource.cpp

```diff
@@ -86,31 +86,25 @@ namespace hiptensor
         // check buffer for C
         auto requiredElementCountC = getProduct(outputSizes);
         auto requiredMemorySizeC   = requiredElementCountC * hipDataTypeSize(dataType);
-        auto needFillDataC         = false;
         if(requiredMemorySizeC > mCurrentAllocByteC)
         {
             Base::reallocDeviceHostPair(mDeviceC, mHostC, requiredMemorySizeC);
             Base::reallocDeviceHostPair(mDeviceReference, mHostReference, requiredMemorySizeC);
             mCurrentAllocByteC = requiredMemorySizeC;
-            needFillDataC      = true;
-        }
-        if(mCurrentDataType != dataType || mCurrentMatrixCElement < requiredElementCountC)
-        {
-            needFillDataC = true;
         }
         mCurrentMatrixCElement = requiredElementCountC;
 
         mCurrentDataType = dataType;
 
+        const uint32_t seedA = 256;
+        const uint32_t seedC = 256;
         if(needFillDataA)
         {
-            fillRand(hostA(), deviceA(), getCurrentMatrixAMemorySize());
+            fillRand(hostA(), deviceA(), dataType, getCurrentMatrixAElement(), seedA);
         }
 
-        if(needFillDataC)
-        {
-            fillRand(hostC(), deviceC(), getCurrentMatrixCMemorySize());
-        }
+        fillRand(hostC(), deviceC(), dataType, getCurrentMatrixCElement(), seedC);
```

Comment:
fixed another bug here. Input C should be reset for each test case since it was overwritten in the last case. 

C = alpah * reduce(A) + beta * C